src/collapse/perf.rs

```diff
@@ -232,12 +232,10 @@ impl Folder {
             }
         }
 
-        if self.in_event || self.skip_stack || !self.stack.is_empty() {
-            return Err(io::Error::new(
-                io::ErrorKind::InvalidData,
-                "Input data ends in the middle of a stack.",
-            ));
-        }
+        // Reset state
```

Comment:
I think I'd like to see an explanation in a comment of _why_ the state has to be reset here. Means the next time someone looks at this code, they don't have to go through all the same thinking you just did :)